components/eamxx/src/physics/nudging/atmosphere_nudging.cpp

```diff
@@ -0,0 +1,275 @@
+  host_views["T_mid"] = view_1d_host<Real>(T_mid_h.data(),T_mid_h.size());
+  layouts.emplace("T_mid", scalar3d_layout_mid);
+
+  fields_ext["p_mid"] = view_2d<Real>("p_mid",m_num_cols,m_num_src_levs);
+  auto p_mid_h       = Kokkos::create_mirror_view(fields_ext["p_mid"]);
+  host_views["p_mid"] = view_1d_host<Real>(p_mid_h.data(),p_mid_h.size());
+  layouts.emplace("p_mid", scalar3d_layout_mid);
+  
+  fields_ext["qv"] = view_2d<Real>("qv",m_num_cols,m_num_src_levs);
+  auto qv_h       = Kokkos::create_mirror_view(fields_ext["qv"]);
+  host_views["qv"] = view_1d_host<Real>(qv_h.data(),qv_h.size());
+  layouts.emplace("qv", scalar3d_layout_mid);
+
+  fields_ext_3d["horiz_winds"] = view_3d<Real>("horiz_winds",m_num_cols,2,m_num_src_levs);
+  auto horiz_winds_h       = Kokkos::create_mirror_view(fields_ext_3d["horiz_winds"]);
+  host_views["horiz_winds"] = view_1d_host<Real>(horiz_winds_h.data(),horiz_winds_h.size());
+  layouts.emplace("horiz_winds", horiz_wind_layout);
+  
+  auto grid_l = m_grid->clone("Point Grid", false);
+  grid_l->reset_num_vertical_lev(m_num_src_levs);
+  ekat::ParameterList data_in_params;
+  m_fnames = {"T_mid","p_mid","qv","horiz_winds"};
+  data_in_params.set("Field Names",m_fnames);
+  data_in_params.set("Filename",datafile);
+  data_in_params.set("Skip_Grid_Checks",true);  // We need to skip grid checks because multiple ranks may want the same column of source data.
+  AtmosphereInput data_input0(m_comm,data_in_params);
+  data_input0.init(grid_l,host_views,layouts);
+  data_input=data_input0;
+
+  T_mid_ext = fields_ext["T_mid"];
+  p_mid_ext = fields_ext["p_mid"];
+  horiz_winds_ext = fields_ext_3d["horiz_winds"];
+  qv_ext = fields_ext["qv"];
+  ts0=timestamp();
+
+  //Load first data for before and after
+  data_input.read_variables(0);
+  NudgingData_bef.init(m_num_cols,m_num_src_levs,true);
+  NudgingData_bef.time = -999;
+  
+  data_input.read_variables(1);
+  NudgingData_aft.init(m_num_cols,m_num_src_levs,true);
+  NudgingData_aft.time = -999;
+}
+
+void Nudging::time_interpolation (const int time_s) {
+
+  using KT = KokkosTypes<DefaultDevice>;
+  using ExeSpace = typename KT::ExeSpace;
+  using ESU = ekat::ExeSpaceUtils<ExeSpace>;
+  using MemberType = typename KT::MemberType;
+  
+  const int time_index = time_s/time_step_file;
+  double time_step_file_d = time_step_file;
+  double w_bef = ((time_index+1)*time_step_file-time_s) / time_step_file_d;
+  double w_aft = (time_s-(time_index)*time_step_file) / time_step_file_d;
+  
+  const int num_cols = NudgingData_aft.T_mid.extent(0);
+  const int num_vert_packs = NudgingData_aft.T_mid.extent(1);
+  const auto policy = ESU::get_default_team_policy(num_cols, num_vert_packs);
+  Kokkos::parallel_for("nudging_time_interpolation", policy,
+     	       KOKKOS_LAMBDA(MemberType const& team) {
+      const int icol = team.league_rank();
+      auto T_mid_1d = ekat::subview(T_mid_ext,icol);
+      auto T_mid_bef_1d = ekat::subview(NudgingData_bef.T_mid,icol);
```

Comment:
All accesses to `NudgingData_xyz` are accessing members of this class. You are not technically allowed to do this inside a `KOKKOS_LAMBDA`. If/when we upgrade to C++17, you can use `KOKKOS_CLASS_LAMBDA`, but until then, you need to do one of the following:
a) use a functor
b) create local copies of `NudgingData_xyz` (as well as any other class-member) right before the pfor call (views are cheap to copy, so that's not expensive).

Reply:
Ok, thanks for the details. I have gone for the local copy route by adding this before the pfor call:
  auto T_mid_bef = view_2d<Real>("",NudgingData_bef.T_mid.extent_int(0),                               NudgingData_bef.T_mid.extent_int(1));
  Kokkos::deep_copy(T_mid_bef,NudgingData_bef.T_mid);
And similar for the other data.

Reply:
Ah, sorry, I should have followed up on this. By copy I meant shallow copy, not deep copy. So
```
auto T_mid_bef = NudgingData_bef.T_mid;
```
is perfect to get around the `this` issue of `KOKKOS_LAMBDA`.

Reply:
I don't think this has been addressed yet.

Reply:
I have updated this, thanks